features/steps/test_post_endpoint.rb

```diff
@@ -1,14 +1,26 @@
 class Spinach::Features::TestPostEndpoint < Spinach::FeatureSteps
   step "I create an Item" do
-    HTTParty.post("http://localhost:3000/api/v1/items", {body: {name: "Foo", quality: 4, sellIn: 4}})
+    options = {
+      body: {name: "Foo", quality: 4, sellIn: 4},
+      headers: {authorization: "Basic Z3VpbGRlZF9yb3NlX2FkbWluOnN1cGVyX3NlY3JldF9wYXNzd29yZF8xMjM="}
```

Comment:
Request for change: If you include the string "Z3Vpb..." here, someone could paste that into a Base64 decoder and get the username / password which was supposed to be a secret. You have a credential leak! In fact, try it out using this link: https://www.base64decode.org/

I would recommend creating a helper method like this:

```
def base64encode(username, password)
  SomeRailsMethod.base_64_encode("#{username}:#{password}")
end
```

And then passing in the secrets:

```
base64encode(Rails.application.credentials.username, Rails.application.credentials.password)
```

Reply:
helpful!